README.md
<p align="center">
  <a href="https://www.lucenthemes.com/">
    <img alt="Sage" src="https://user-images.githubusercontent.com/37496983/115125341-1eeb5b00-9fe5-11eb-923e-886e1dda3e28.png" height="300">
  </a>
</p>
<p align="center">
    <a href="LICENSE.md"><img alt="GPL License" src="https://img.shields.io/github/license/naveenkharwar/agilitywp?label=Proudly%20Open%20Source"></a>
    <img alt="GitHub package.json version" src="https://img.shields.io/github/package-json/v/naveenkharwar/agilitywp">
    <img alt="GitHub commit activity" src="https://img.shields.io/github/commit-activity/w/naveenkharwar/agilitywp">
</p>

<p align="center">
  <strong>Just Another Fantastic WordPress Theme for your new WordPress website built by some amazing young thoughts from the community.</strong>
  <br />
  Built with ❤️
</p>

<p align="center">
  <a href="https://www.lucenthemes.com/agilitywp">Official Website</a> | <a href="https://www.lucenthemes.com/docs/agilitywp/">Documentation</a> | <a href="CHANGELOG.md">Change Log</a>
</p>

## ✨ About Agility WP

**Agility WP** is a WordPress Theme that is designed for real case scenarios. Agility is following some world-class UI and UX Rules to best fit for Bloggers, E-Commerce, Marketing, Portfolio websites.

Its a is highly customizable and SEO Friendly and is tested for Good SEO and Performance. We focus more to make this Theme Lightweight, SEO Friendly and blazing fast. The theme is having a clean UI that attracts many of its users.

We did a good job of making Agility WP Responsive, Cross-Browser Support, WooCommerce Ready, Gutenberg, Eleementor, DIVI, Beaver Ready, Always UP TO DATE concerning current WordPress Version.

Looking for an amazing WordPress Theme for your new website which has 24x7 Support and continuous improvement?
Welcome to LucenWorld.

### 📦 Requirements

Make sure all dependencies have been installed before moving on:

- [WordPress](https://wordpress.org/) >= 5.4
- [PHP](https://secure.php.net/manual/en/install.php) >= 7.5.0
- [Composer](https://getcomposer.org/download/)
- [Node.js](http://nodejs.org/) >= 12.14.0

### 🚀  Theme installation

**From within WordPress**

1. Visit "Appearance > Themes > Add New"
2. Search for "Agility WP"
3. Install and activate

**Using FTP**

1. Download [Agility WP](https://www.wordpress.org/themes/agilitywp)
2. Using an FTP client to access your host web server and navigate to /wp-content/themes/ directory.
3. Upload the Theme folder (agilitywp.zip) to this directory on your host server.
4. Login to WordPress Dashboard.
5. Select the Appearance screen, then Themes.
6. Look for Agility WP and Activate it.

### ⚙️ Theme development

```sh
git clone https://github.com/NaveenKharwar/agilitywp
cd agilitywp
npm install # To install all package
```

### 👨🏼‍💻 Build commands

- `grunt watch` — Compile assets when file changes are made.
- `grunt build` — Compile and optimize the files in your assets directory
- `grunt prod` — Compile assets for production and generate `build/agilitywp.zip`.

### 📖 Documentation

- ~~[Sage documentation](https://www.lucenthemes.com/docs/agilitywp/)~~

### 🙌 Contributers

| [Naveen Kharwar \ Developer](https://twitter.com/naveenkharwar21) | [LucenThemes \ Maintainer](https://www.lucenthemes.com/)  |
| :------------: | :------------: |
|  [<img alt="naveenkharwar" src="https://avatars.githubusercontent.com/u/37496983?v=4" width="200">](https://twitter.com/naveenkharwar21) | [<img alt="LucenThemes" src="https://user-images.githubusercontent.com/37496983/88413968-c0000900-cdf9-11ea-8d93-017b54823510.png" width="200">](https://www.lucenthemes.com/)  |

### 🤝 Agility WP sponsors

Help support our open-source development efforts by ~~[becoming a sponsor]~~(https://github.com/sponsors/agilitywp).

### 🏚 Community

Keep track of development and community news.


CHANGELOG.md
### 0.0.3
- Modify Navigation
- Fix Accessibility

### 0.0.2
- Add Style editor
- Gutenberg Compatible
- Categories badge fix
- Screenshot Change
- Change Site logo size

### v0.01
- Initial Release

functions.php
<?php
/**
 * Agility WP functions and definitions
 *
 * @link https://developer.wordpress.org/themes/basics/theme-functions/
 *
 * @package Agility_WP
 */

if ( ! defined( 'AGILITYWP_VERSION' ) ) {
	// Replace the version number of the theme on each release.
	define( 'AGILITYWP_VERSION', '0.0.3' );
}

if ( ! function_exists( 'AGILITYWP_VERSION' ) ) :
	/**
	 * Sets up theme defaults and registers support for various WordPress features.
	 *
	 * Note that this function is hooked into the after_setup_theme hook, which
	 * runs before the init hook. The init hook is too late for some features, such
	 * as indicating support for post thumbnails.
	 */
	function agilitywp_setup() {
		/*
		 * Make theme available for translation.
		 * Translations can be filed in the /languages/ directory.
		 * If you're building a theme based on Agility WP, use a find and replace
		 * to change 'agilitywp' to the name of your theme in all the template files.
		 */
		load_theme_textdomain( 'agilitywp', get_template_directory() . '/languages' );

		// Add default posts and comments RSS feed links to head.
		add_theme_support( 'automatic-feed-links' );

		/*
		 * Let WordPress manage the document title.
		 * By adding theme support, we declare that this theme does not use a
		 * hard-coded <title> tag in the document head, and expect WordPress to
		 * provide it for us.
		 */
		add_theme_support( 'title-tag' );

		/*
		 * Enable support for Post Thumbnails on posts and pages.
		 *
		 * @link https://developer.wordpress.org/themes/functionality/featured-images-post-thumbnails/
		 */
		add_theme_support( 'post-thumbnails' );

		// This theme uses wp_nav_menu() in one location.
		register_nav_menus(
			array(
				'primary' => esc_html__( 'Primary', 'agilitywp' ),
			)
		);

		/*
		 * Switch default core markup for search form, comment form, and comments
		 * to output valid HTML5.
		 */
		add_theme_support(
			'html5',
			array(
				'search-form',
				'comment-form',
				'comment-list',
				'gallery',
				'caption',
				'style',
				'script',
			)
		);

		// Set up the WordPress core custom background feature.
		add_theme_support(
			'custom-background',
			apply_filters(
				'agilitywp_custom_background_args',
				array(
					'default-color' => 'ffffff',
					'default-image' => '',
				)
			)
		);

		// Add theme support for selective refresh for widgets.
		add_theme_support( 'customize-selective-refresh-widgets' );

		// Add Theme Support for Editor Style.
		add_theme_support( 'editor-styles' );

		// Add Theme Support for Block Styles.
		add_theme_support( 'wp-block-styles' );

		// Add Theme Support for Wide and FFull alignment.
		add_theme_support( 'align-wide' );

		// Add Support for responsive embed.
		add_theme_support( 'responsive-embeds' );

		/**
		 * Add support for core custom logo.
		 *
		 * @link https://codex.wordpress.org/Theme_Logo
		 */
		add_theme_support(
			'custom-logo',
			array(
				'height'      => 250,
				'width'       => 250,
				'flex-width'  => true,
				'flex-height' => true,
			)
		);
	}
endif;
add_action( 'after_setup_theme', 'agilitywp_setup' );

/**
 * Set the content width in pixels, based on the theme's design and stylesheet.
 *
 * Priority 0 to make it available to lower priority callbacks.
 *
 * @global int $content_width
 */
function agilitywp_content_width() {
	// This variable is intended to be overruled from themes.
	// Open WPCS issue: {@link https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/1043}.
	// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedVariableFound
	$GLOBALS['content_width'] = apply_filters( 'agilitywp_content_width', 640 );
}
add_action( 'after_setup_theme', 'agilitywp_content_width', 0 );

/**
 * Register widget area.
 *
 * @link https://developer.wordpress.org/themes/functionality/sidebars/#registering-a-sidebar
 */
function agilitywp_widgets_init() {
	register_sidebar(
		array(
			'name'          => esc_html__( 'Sidebar', 'agilitywp' ),
			'id'            => 'sidebar-1',
			'description'   => esc_html__( 'Add widgets here.', 'agilitywp' ),
			'before_widget' => '<section id="%1$s" class="sidebar-block widget %2$s">',
			'after_widget'  => '</section>',
			'before_title'  => '<h6 class="widget-title">',
			'after_title'   => '</h6>',
		)
	);
	register_sidebar(
		array(
			'name'          => esc_html__( 'Footer Widget', 'agilitywp' ),
			'id'            => 'footer-widget',
			'description'   => esc_html__( 'Add widgets here.', 'agilitywp' ),
			'before_widget' => '<div id="%1$s" class="col-lg-3 col-md-6 col-12  widget %2$s">',
			'after_widget'  => '</div>',
			'before_title'  => '<h6 class="widget-title"><span>',
			'after_title'   => '</span></h6>',
		)
	);
	register_sidebar(
		array(
			'name'          => esc_html__( 'Footer Bottom Widget', 'agilitywp' ),
			'id'            => 'footer_bottom_widget',
			'description'   => esc_html__( 'Add widgets here.', 'agilitywp' ),
			'before_widget' => '<div id="%1$s" class="widget %2$s">',
			'after_widget'  => '</div>',
			'before_title'  => '<h6 class="widget-title"><span>',
			'after_title'   => '</span></h6>',
		)
	);
}
add_action( 'widgets_init', 'agilitywp_widgets_init' );

/**
 * Enqueue styles for Editor.
 */
function agilitywp_block_editor_styles() {
	wp_enqueue_style( 'agilitywp-editor-styles', get_theme_file_uri( 'assets/css/style-editor.css' ), false, AGILITYWP_VERSION, 'all' );
}
add_action( 'enqueue_block_editor_assets', 'agilitywp_block_editor_styles' );

/**
 * Enqueue scripts and styles.
 */
function agilitywp_scripts() {
	wp_enqueue_style( 'agilitywp-style', get_stylesheet_uri(), array(), AGILITYWP_VERSION );
	wp_style_add_data( 'agilitywp-style', 'rtl', 'replace' );
	wp_enqueue_style( 'agilitywp-theme', get_template_directory_uri() . '/assets/css/theme.min.css', array(), AGILITYWP_VERSION );

	wp_enqueue_style( 'agilitywp-catamaran-font', '//fonts.googleapis.com/css2?family=Catamaran:wght@100;200;300;400;500;600;700;800;900&display=swap', array(), AGILITYWP_VERSION );
	wp_enqueue_style( 'agilitywp-quicksend-font', '//fonts.googleapis.com/css2?family=Quicksand:wght@300;400;500;600;700&display=swap', array(), AGILITYWP_VERSION );

	if ( is_singular() && comments_open() && get_option( 'thread_comments' ) ) {
		wp_enqueue_script( 'comment-reply' );
	}

	wp_enqueue_script( 'agilitywp-bootstrap-js', get_template_directory_uri() . '/assets/js/bootstrap.min.js', array(), AGILITYWP_VERSION, true );
}
add_action( 'wp_enqueue_scripts', 'agilitywp_scripts' );

/**
 * Custom Header for this theme.
 */
require get_template_directory() . '/inc/custom-header.php';

/**
 * Custom template tags for this theme.
 */
require get_template_directory() . '/inc/template-tags.php';

/**
 * Functions which enhance the theme by hooking into WordPress.
 */
require get_template_directory() . '/inc/template-functions.php';

/**
 * Customizer additions.
 */
require get_template_directory() . '/inc/customizer.php';

/**
 * Load Jetpack compatibility file.
 */
if ( defined( 'JETPACK__VERSION' ) ) {
	require get_template_directory() . '/inc/jetpack.php';
}

/**
 * Load WooCommerce compatibility file.
 */
if ( class_exists( 'WooCommerce' ) ) {
	require get_template_directory() . '/inc/woocommerce.php';
}

if ( ! file_exists( get_template_directory() . '/inc/class-wp-bootstrap-navwalker.php' ) ) {
	return new WP_Error( 'class-wp-bootstrap-navwalker-missing', __( 'It appears the class-wp-bootstrap-navwalker.php file may be missing.', 'agilitywp' ) );
} else {
	require_once get_template_directory() . '/inc/class-wp-bootstrap-navwalker.php';
}

/**
 * Replaces the excerpt "Read More" text by a link.
 *
 * @param string $more for blog page.
 * @return Post Link
 */
function agilitywp_excerpt_more( $more ) {
	return sprintf(
		'<p class="text-end"><a href="%1$s" class="ead-more_tag">%2$s</a></p>',
		esc_url( get_permalink( get_the_ID() ) ),
		/* translators: %s: arrow icon */
		sprintf( __( 'Continue reading %s', 'agilitywp' ), '<span> &#8594; </span>' )
	);
}
add_filter( 'excerpt_more', 'agilitywp_excerpt_more' );
